cmd/encryption-v1.go

```diff
@@ -756,19 +742,22 @@ func (o *ObjectInfo) DecryptedSize() (int64, error) {
 	if !o.IsEncrypted() {
 		panic("cannot compute decrypted size of an object which is not encrypted")
 	}
-
-	return decryptedSize(o.Size)
+	size, err := sio.DecryptedSize(uint64(o.Size))
+	if err != nil {
+		err = errObjectTampered // assign correct error type
+	}
+	return int64(size), err
 }
 
 // EncryptedSize returns the size of the object after encryption.
 // An encrypted object is always larger than a plain object
 // except for zero size objects.
 func (o *ObjectInfo) EncryptedSize() int64 {
-	size := (o.Size / sseDAREPackageBlockSize) * (sseDAREPackageBlockSize + sseDAREPackageMetaSize)
-	if mod := o.Size % (sseDAREPackageBlockSize); mod > 0 {
-		size += mod + sseDAREPackageMetaSize
+	size, err := sio.EncryptedSize(uint64(o.Size))
+	if err != nil {
+		panic(err) // Since AWS S3 allows parts to be 5GB at most this cannot happen - sio max. size is 256 TB
```

Comment:
Looks like the server will panic if data in the backend are corrupted. In general we don't like the server to crash. I think it makes sense to also return an error here and make callers check the result. I checked EncryptedSize() is called only few times.

Reply:
Server will not crash only the goroutine to the handler crashes. Go http server survives recoverable panics. This panic will only occur at 256TB and above since it is unlikely a panic here is okay.

Reply:
Unlikely with the sense that we have other checks to prevent object > 5TB

Reply:
oh so that's the only case when EncryptedSize returns an error then